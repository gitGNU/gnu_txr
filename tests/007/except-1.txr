@(defex u0)
@(defex u1)
@(define baz (x))
@  (cases)
@    (bind x "0")
@    (throw u0 "text0")
@  (or)
@    (bind x "1")
@    (throw u1 "text1")
@  (end)
@(end)
@(define bar (x))
@  (baz x)
@(end)
@(define foo ())
@  (next `!echo "0\n1\n"`)
@  (collect)
@num
@    (try)
@      (bar num)
@    (catch u0 (arg))
@      (output)
caught u0: @arg
@      (end)
@    (catch u1 (arg))
@      (output)
caught u1: @arg
@      (end)
@    (end)
@  (end)
@(end)
@(foo)
