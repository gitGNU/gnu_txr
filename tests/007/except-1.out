caught u0: text0
caught u1: text1
